### PR TITLE
Added pattern unit tests for detectors starting with the letters r through s

### PR DESCRIPTION
**Pull request by nabeelalam**
### Description:
This PR adds pattern test cases for detectors starting with the letters r through s.
It also separates non-pattern tests into its integration tests file for those detectors.

### Test Output:
image

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/ramp/ramp_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
```
Comment (ahrav) [inline pkg/detectors/ramp/ramp_integration_test.go:15]: nit: Could we move this import down one line so it's grouped with the others.

File: pkg/detectors/replicate/replicate_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
```
Comment (ahrav) [inline pkg/detectors/replicate/replicate_integration_test.go:15]: ditto

File: pkg/detectors/replyio/replyio_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
```
Comment (ahrav) [inline pkg/detectors/replyio/replyio_integration_test.go:15]: ditto

File: pkg/detectors/requestfinance/requestfinance_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
```
Comment (ahrav) [inline pkg/detectors/requestfinance/requestfinance_integration_test.go:15]: here as well

**Review by ahrav — APPROVED** [review]
Thank you for cleaning all of these up! Just a few minor formatting comments; otherwise, LGTM. 👍

**Review by kashifkhan0771 — APPROVED** [review]

File: pkg/detectors/ramp/ramp_integration_test.go (hunk 1 above)
Comment (nabeelalam) [reply pkg/detectors/ramp/ramp_integration_test.go]: Thanks for pointing that out. I went in and fixed the imports for all the new integration tests.

**Pull request merged**